lib/facter/resolvers/base_resolver.rb

```diff
@@ -18,13 +18,21 @@ def self.subscribe_to_manager
       def self.resolve(fact_name)
         @semaphore.synchronize do
           subscribe_to_manager
+
           post_resolve(fact_name)
```

Comment:
Would it be ok to call post_resolve only if @fact_list is empty?

Reply:
Not really, you loose control on your resolver call if the execution stops in base_resolver. Compared to exceptions that are caught in base_resolver, this is a more common flow that allows you to 'cache' successful resolver calls that have no output.

Reply:
We can have partial results in @fact_list so we cannot decide in the base resolver if we want to call the resolver or not. We should always call the resolver.